packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1221,6 +1223,11 @@ class EditableText extends StatefulWidget {
   ///    its parent.
   final int? maxLines;
 
+
+  /// The maximum number of characters that can be entered in the text field.
+  /// If `null`, there is no maximum limit.
+  final int? maxLength;
```

Comment:
I think it would be better if we don't have to change the public API for this fix.  Is it possible to change the logic over [here](https://github.com/flutter/flutter/blob/3b2652b73534fe2c5e91e277f1349b96660abcd9/packages/flutter/lib/src/widgets/editable_text.dart#L3222-L3229) so the obsured text doesn't get revealed if the text isn't changed?